packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -0,0 +1,239 @@
+import iam = require('@aws-cdk/aws-iam');
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { IDatabase } from './database';
+import { CfnTable } from './glue.generated';
+import { Column } from './schema';
+import { StorageType } from './storage-type';
+
+export interface ITable extends cdk.IConstruct {
+  readonly tableArn: string;
+  readonly tableName: string;
+
+  export(): TableImportProps;
+}
+
+export interface TableImportProps {
+  tableArn: string;
+  tableName: string;
+}
+
+export interface TableProps {
+  /**
+   * Name of the table.
+   */
+  tableName: string;
+
+  /**
+   * Description of the table.
+   *
+   * @default generated
+   */
+  description?: string;
+
+  /**
+   * Database in which to store the table.
+   */
+  database: IDatabase;
+
+  /**
+   * S3 bucket in which to store data.
+   *
+   * @default one is created for you
+   */
+  bucket?: s3.Bucket;
+
+  /**
+   * S3 prefix under which table objects are stored.
+   *
+   * @default data/
+   */
+  prefix?: string;
+
+  /**
+   * Columns of the table.
+   */
+  columns: Column[];
+
+  /**
+   * Partition columns of the table.
+   *
+   * @default table is not partitioned
+   */
+  partitionKeys?: Column[]
+
+  /**
+   * Storage type of the table's data.
+   */
+  storageType: StorageType;
+
+  /**
+   * Indicates whether the table's data is compressed or not.
+   *
+   * @default false
+   */
+  compressed?: boolean;
+
+  /**
+   * Indicates whether the table data is stored in subdirectories.
+   *
+   * @default false
+   */
+  storedAsSubDirectories?: boolean;
+}
+
+/**
+ * A Glue table.
+ */
+export class Table extends cdk.Construct implements ITable {
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param props A `TableImportProps` object. Can be obtained from a call to `table.export()` or manually created.
+   */
+  public static import(scope: cdk.Construct, id: string, props: TableImportProps): ITable {
+    return new ImportedTable(scope, id, props);
+  }
+
+  public readonly database: IDatabase;
+  public readonly bucket: s3.Bucket;
+  public readonly prefix: string;
+
+  public readonly tableName: string;
+  public readonly tableArn: string;
+
+  public readonly storageType: StorageType;
+  public readonly columns: Column[];
+  public readonly partitionKeys?: Column[];
+
+  constructor(scope: cdk.Construct, id: string, props: TableProps) {
+    super(scope, id);
+    this.database = props.database;
+    this.bucket = props.bucket || new s3.Bucket(this, 'Bucket');
```

Comment:
Encryption options? I wonder if we should expose the bucket encryption option for the default bucket:

```ts
new Table(this, 'MyTable', { encryption: xxx })
```